File: hall_of_fame/seunggil/python/1018.py
import sys

n = sys.stdin.readline().rstrip().split()

n = list(map(int,n))
inputData = []
for i in range(n[0]):
    inputData.append(list(sys.stdin.readline().rstrip()))

answer = 1 << 31 - 1
for i in range(n[0]-7):
    for j in range(n[1]-7):
        color = 'W'
        ans = [0,0]
        for a in range(i,i+8):
            for b in range(j,j+8):
                if inputData[a][b] == color:
                    ans[0] +=1
                if inputData[a][b] != color:
                    ans[1] +=1
                color = 'B' if color == 'W' else 'W'
            color = 'B' if color == 'W' else 'W'
        answer = min([answer,min(ans)])
print(answer)

File: hall_of_fame/seunggil/c++/1018.cpp
#include <iostream>
#include <vector>

using namespace std;

vector<vector<bool>> inputData;

int white(int i, int j) {
	bool ans = true;
	int result = 0;
	for (int a = i; a < i + 8; a++) {
		for (int b = j; b < j + 8; b++) {
			if (inputData[a][b] != ans)
				result++;
			ans = !ans;
		}
		ans = !ans;
	}
	return result;
}

int black(int i, int j) {
	bool ans = false;
	int result = 0;
	for (int a = i; a < i + 8; a++) {
		for (int b = j; b < j + 8; b++) {
			if (inputData[a][b] != ans)
				result++;
			ans = !ans;
		}
		ans = !ans;
	}
		
	return result;
}

int main(void) {
	int n, m;
	unsigned int answer = -1;
	cin >> n >> m;

	inputData = vector<vector<bool>>(n, vector<bool>(m));
	
	for(int i = 0; i <n; i++)
		for (int j = 0; j < m; j++) {
			char c;
			cin >> c;
			inputData[i][j] = c == 'W' ? true : false;
		}

	for (int i = 0; i+7 < n; i++)
		for (int j = 0; j+7 < m; j++) {
			int result = min(white(i, j), black(i, j));

			if (answer > result)
				answer = result;	
		}

	cout << answer;
	
	return 0;
}